addOns/retest/src/main/java/org/zaproxy/addon/retest/EditAlertPanel.java

```diff
@@ -0,0 +1,365 @@
+    }
+
+    private JScrollPane createJScrollPane(String name) {
+        JScrollPane jScrollPane = new JScrollPane();
+        jScrollPane.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        jScrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED);
+        jScrollPane.setBorder(
+                BorderFactory.createTitledBorder(
+                        null,
+                        name,
+                        TitledBorder.DEFAULT_JUSTIFICATION,
+                        javax.swing.border.TitledBorder.DEFAULT_POSITION,
+                        FontUtils.getFont(FontUtils.Size.standard)));
+        return jScrollPane;
+    }
+
+    private JPanel getAlertDisplay() {
+        if (alertDisplay == null) {
+            alertDisplay = new JXPanel();
+            alertDisplay.setLayout(new GridBagLayout());
+            alertDisplay.setScrollableHeightHint(ScrollableSizeHint.NONE);
+            alertDisplay.setName("alertDisplay");
+
+            alertEditPlugin = new ZapNumberSpinner();
+            if (alertEditPlugin.getEditor() instanceof JSpinner.DefaultEditor) {
+                ((JSpinner.DefaultEditor) alertEditPlugin.getEditor())
+                        .getTextField()
+                        .setHorizontalAlignment(JTextField.LEFT);
+            }
+
+            alertEditName = new JComboBox<>();
+            alertEditName.setEditable(true);
+            nameListModel = new DefaultComboBoxModel<>();
+            List<String> allVulns = getAllVulnerabilityNames();
+            for (String vuln : allVulns) {
+                nameListModel.addElement(vuln);
+            }
+            alertEditName.setModel(nameListModel);
+
+            alertEditMethod = new JComboBox<>();
+            alertEditMethod.setEditable(true);
+            methodListModel = new DefaultComboBoxModel<>();
+            String[] allMethods = HttpRequestHeader.METHODS;
+            for (String method : allMethods) {
+                methodListModel.addElement(method);
+            }
+            alertEditMethod.setModel(methodListModel);
+
+            alertEditUrl = new ZapTextField();
+            alertEditAttack = new ZapTextField();
+
+            paramListModel = new DefaultComboBoxModel<>();
+            paramListModel.addElement(""); // Default is empty so user can type anything in
+            alertEditParam = new JComboBox<>();
+            alertEditParam.setModel(paramListModel);
+            alertEditParam.setEditable(true);
+
+            alertEditEvidence = new ZapTextField();
+            alertEditConfidence = new JComboBox<>(Alert.MSG_CONFIDENCE);
+            alertEditRisk = new JComboBox<>(Alert.MSG_RISK);
+
+            alertOtherInfo = createZapTextArea();
+            JScrollPane otherSp =
+                    createJScrollPane(Constant.messages.getString("alert.label.other"));
+            otherSp.setViewportView(alertOtherInfo);
+
+            int gbcRow = 0;
+
+            alertDisplay.add(
```

Comment:
Labels should be associated with the input component that they are associated with - this is for accessibility. To do this use the "jlabel.setLabelFor(..)" method - the StandardField dialogs do this for you :)